src/include/sof/drivers/dmic.h

```diff
@@ -315,17 +316,33 @@
 #define dmic_irq(dmic) dmic->plat_data.irq
 #define dmic_irq_name(dmic) dmic->plat_data.irq_name
 
-/* DMIC private data */
-struct dmic_pdata {
-	uint16_t enable[DMIC_HW_CONTROLLERS];
-	uint32_t state;
+struct dmic_private_nonshared {
 	struct task dmicwork;
 	int32_t startcount;
 	int32_t gain;
-	int32_t gain_coef;
+};
+
+struct dmic_private_shared {
+	uint16_t enable[DMIC_HW_CONTROLLERS];	/* set_config() on core 0 */
```

Comment:
I thought this structure was per DMIC instance, shouldn't this array be in `dmic_global_shared` below? In fact I'm confused: Sue Creek is the only platform where `DMIC_HW_CONTROLLERS != DMIC_HW_FIFOS`. Probing seems to be done per FIFO, not per controller. So on platforms like Sue Creek, which has 4 controllers and 2 FIFOs, controllers share FIFOs? Or does this array mean - which controllers are enabled for *this* FIFO?

Reply:
I added comments to structs in dmic.h, that should help.